DDL.sql

```diff
@@ -0,0 +1,263 @@
+    totalVoltageUnits TEXT DEFAULT NULL,
+    batteryVoltageUnits TEXT DEFAULT NULL,
+    currentUnits TEXT DEFAULT NULL,
+    thermistorTempUnits TEXT DEFAULT NULL,
+    packTempUnits TEXT DEFAULT NULL,
+    bqTempUnits TEXT DEFAULT NULL,
+    cellVoltageUnits TEXT DEFAULT NULL
+);
+
+CREATE TABLE IMUConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    vectorUnits TEXT DEFAULT NULL
+);
+
+CREATE TABLE TMUConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    thermUnits TEXT NOT NULL
+);
+
+CREATE TABLE TMSConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL,
+    sensorTempUnits TEXT NOT NULL,
+    pumpSpeedUnits TEXT NOT NULL,
+    fanSpeedUnits TEXT NOT NULL
+);
+
+CREATE TABLE PVCConfig(
+    ID SERIAL PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE TABLE BikeBoardConfig(
+    bikeConfigID INTEGER,
+    boardConfig INTEGER,
+    PRIMARY KEY ( bikeConfigID, boardConfig)
+);
+
+CREATE TABLE BikePartConfig(
+    ID SERIAL PRIMARY KEY,
+    bikeConfigId INTEGER NOT NULL,
+    partNumber TEXT NOT NULL
+);
+
+CREATE TABLE Context(
+    ID SERIAL PRIMARY KEY,
+    date DATE NOT NULL,
+    runEventType TEXT NOT NULL,
+    airTemp INTEGER DEFAULT NULL,
+    altitude INTEGER DEFAULT NULL,
+    location TEXT NOT NULL,
+    windVelocity FLOAT DEFAULT NULL,
+    riderOne TEXT NOT NULL,
+    riderTwo TEXT DEFAULT NULL,
+    riderOneCurrentWeight FLOAT NOT NULL,
+    riderTwoCurrentWeight FLOAT DEFAULT NULL,
+    humidity FLOAT DEFAULT NULL,
+    driverFeedback TEXT DEFAULT NULL
+);
+
+CREATE TABLE BMSNumAttemptsMade(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSNumThermAttemptsMade(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL
+);
+
+CREATE TABLE BMSLastAttemptTime(
+    ID SERIAL PRIMARY KEY,
+    time INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSThermAttemptTime(
+    ID SERIAL PRIMARY KEY,
+    time INTEGER NOT NULL,
+    timeStamp TIMESTAMP NOT NULL,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSTotalVoltage(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSBatteryVoltage(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSCurrent(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSThermistorTemp(
+
+    ID SERIAL,
+    thermNumber INTEGER,
+    PRIMARY KEY (ID, thermNumber),
+    value INTEGER NOT NULL,
+    timeStamp TIMESTAMP,
+    context_id INTEGER REFERENCES Context
+);
+
+CREATE TABLE BMSMinPackTemp(
+    ID SERIAL PRIMARY KEY,
+    value INTEGER,
+    packId INTEGER,
```

Comment:
The ID that the CAN message is reporting is the ID of the thermistor inside the pack, not of the pack itself. That said, we are probably going to have to store an identifier for each of the packs in all the BMS tables, so we can identify which pack the data is from. From a user perspective, we identify the packs as "1A", "1B",... "3B", so you could store this ID as a 2 character string